lib/protocol/packets/RowDataPacket.js

```diff
@@ -61,7 +61,7 @@ function typeCast(field, parser, timeZone, supportBigNumbers, bigNumberStrings,
     case Types.DATETIME2:
     case Types.NEWDATE:
       var dateString = parser.parseLengthCodedString();
-      if (dateStrings) {
+      if (dateStrings === true || field.type === Types.DATE && dateStrings.toString().toLowerCase() === "date") {
```

Comment:
Thanks for the feedback, I'll look at supporting any truthy value - except `dateStrings.toString().toLowerCase() === "date"` - for backward compatible behavior and also look to adding the requested tests.
I may be back to ask for help on setting up the test.